### PR TITLE
Preliminary docs for Datastore and Logging

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -35,3 +35,39 @@ Each abstraction is built over the lower-level one, and client code
...
+    Key = new Key { PartitionId = partitionId, Path = { new PathElement { Kind = "message" } } },
+    ["created"] = DateTime.UtcNow,
+    ["text"] = "Text of the message"
+};
+var transaction = client.BeginTransaction(projectId).Transaction;
+var commitResponse = client.Commit(projectId, Mode.TRANSACTIONAL, transaction, new[] { entity.ToInsert() });
+var insertedKey = commitResponse.MutationResults[0].Key;
+```
+
+See [`DatastoreClient`](obj/api/Google.Datastore.V1Beta3.DatastoreClient.yml)
```
Comment (chrisdunelm) [inline docs/index.md]: What are these links? Do they work?
The existing links currently on github for pubsub 404 for me.

Comment (jskeet) [inline docs/index.md]: They should work, yes. Which links fail at the moment, exactly? I've just repushed http://googlecloudplatform.github.io/gcloud-dotnet/ - see if it works for you now.

Comment (chrisdunelm) [inline docs/index.md]: All the .yml links on https://github.com/GoogleCloudPlatform/gcloud-dotnet/blob/master/docs/index.md fail.
E.g.
https://github.com/GoogleCloudPlatform/gcloud-dotnet/blob/master/docs/obj/api/Google.Pubsub.V1.PublisherClient.yml
https://github.com/GoogleCloudPlatform/gcloud-dotnet/blob/master/docs/obj/api/Google.Pubsub.V1.Subscriber.ISubscriberClient.yml

Comment (jskeet) [inline docs/index.md]: Ah, yes - they'll fail while they're being view as just source. They work when they're rendered by docfx :) I'm not sure there's a nice way round that...

Comment (chrisdunelm) [inline docs/index.md]: OK then, LGTM

**Pull request merged**